linkis-commons/linkis-common/src/main/java/org/apache/linkis/common/utils/SecurityUtils.java

```diff
@@ -0,0 +1,193 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.linkis.common.utils;
+
+import org.apache.linkis.common.conf.CommonVars;
+import org.apache.linkis.common.conf.CommonVars$;
+import org.apache.linkis.common.exception.LinkisSecurityException;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class SecurityUtils {
+
+  private static final Logger logger = LoggerFactory.getLogger(SecurityUtils.class);
+
+  private static final String COMMA = ",";
+
+  private static final String EQUAL_SIGN = "=";
+
+  private static final String AND_SYMBOL = "&";
+
+  private static final String QUESTION_MARK = "?";
+
+  /** allowLoadLocalInfile,allowLoadLocalInfiled,# */
+  public static final CommonVars<String> MYSQL_SENSITIVE_PARAMS =
+      CommonVars$.MODULE$.apply(
+          "linkis.mysql.sensitive.params",
+          "allowLoadLocalInfile,autoDeserialize,allowLocalInfile,allowUrlInLocalInfile,#");
+
+  /**
+   * "allowLoadLocalInfile=false&autoDeserialize=false&allowLocalInfile=false&allowUrlInLocalInfile=false"
+   */
+  public static final CommonVars<String> MYSQL_FORCE_PARAMS =
+      CommonVars$.MODULE$.apply(
+          "linkis.mysql.force.params",
+          "allowLoadLocalInfile=false&autoDeserialize=false&allowLocalInfile=false&allowUrlInLocalInfile=false");
+
+  public static final CommonVars<String> MYSQL_STRONG_SECURITY_ENABLE =
+      CommonVars$.MODULE$.apply("linkis.mysql.strong.security.enable", "false");
+
+  /**
+   * mysql url append force params
+   *
+   * @param url
+   * @return
+   */
+  public static String appendMysqlForceParams(String url) {
+    if (StringUtils.isBlank(url)) {
+      return "";
+    }
+    Map<String, Object> forceParams = parseMysqlUrlParamsToMap(MYSQL_FORCE_PARAMS.getValue());
+
+    String extraParamString = parseParamsMapToMysqlParamUrl(forceParams);
+
+    if (url.endsWith(QUESTION_MARK)) {
+      url = url + extraParamString;
+    } else if (url.lastIndexOf(QUESTION_MARK) < 0) {
+      url = url + QUESTION_MARK + extraParamString;
+    } else {
+      url = url + AND_SYMBOL + extraParamString;
+    }
+    return url;
+  }
+
+  public static void appendMysqlForceParams(Map<String, Object> extraParams) {
+    extraParams.putAll(parseMysqlUrlParamsToMap(MYSQL_FORCE_PARAMS.getValue()));
+  }
+
+  public static String checkJdbcSecurity(String url) {
```

Comment:
checkJdbcSecurity should clearSensitiveParams

Reply:
Throw an exception directly if there are sensitive parameters.